docs/source/app-dev/app-arch.rst

```diff
@@ -246,3 +235,79 @@ There are some other identifiers that are determined by your client code. These
 .. |image0| image:: images/BotFlow.png
    :width: 6.5in
    :height: 3.69444in
+
+Testing
+=======
+
+Testing is fundamental to ensure correctness and favor maintainability.
+
+Testing is usually divided in different categories according to its scope and aim:
+
+- unit testing verifies single properties of individual components
+- integration testing verifies that an aggregation of components behaves as expected
+- acceptance testing checks that the overall behavior of a whole system satisfies certain criteria
+
+Both tests in the small scale (unit testing) and large (acceptance testing) tend to be specific to the given component or system under test.
+
+This chapter focuses on providing portable approaches and techniques to perform integration testing between your components and an actual running ledger.
+
+Test the business logic with a ledger
+*************************************
+
+In production, your application is going to interact with a DAML model deployed on an actual ledger.
+
+Each model is usually specific to a business need and describes specific protocols.
+
+Mocking a ledger response is usually not desirable to test the business logic, because so much of it is encapsulated in the DAML model.
+
+This makes integration testing with an actual running ledger fundamental to evaluating the correctness of an application.
+
+This is usually achieved by running a ledger as part of the test process and run several tests against it, possibly coordinated by a test framework.
+
+Since the in-memory sandbox shipped as part of the SDK is a full-fledged implementation of a DAML ledger, it's usually the tool of choice for these tests.
+
+Please note that this does not replace acceptance tests with the actual ledger implementation that your application aims to use in production.
+
+Whatever your choice is, sharing a single ledger to run several tests is a suggested best practice.
+
+Share the ledger
```

Comment:
This is too much text for the point you are making. Maybe combine with the next section and just add a single paragraph there to motivate sharing the ledger.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/2505/commits/319dad7e7f104cce9f6556e5c93247d4656a7556